woocommerce/woocommerce-android
# Issue/2844 order detail refactor order notes card

This PR takes care of the fifth step of the Order Detail refactoring #2844 .

#### Changes
- Adds the Order Notes card (`OrderDetailOrderNotesView`) to the Order detail screen.
- Adds support to fetch and display order notes from FluxC.
- Adds logic to pass `localOrderId` when opening Order detail screen. Note that clicking on a notification and opening the Order Detail screen is not yet implemented/tested and will take place in another PR.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- This PR just adds the Order detail Payment info card to the Order detail screen. The remaining cards will be added in subsequent PRs along with unit tests and click actions for all the buttons in the Order detail.
- **This PR is in draft till this [Step 4 PR](https://github.com/woocommerce/woocommerce-android/pull/2861) can be merged.**

#### Testing
- Click on an order from the Orders tab.
- Verify that the Order notes skeleton view is displayed and the Order notes are displayed once the notes are fetched.. **Note that the `Add note` section of the `Notes` card will take place in a subsequent PR as mentioned in the GitHub issue**

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/93609469-8870a000-f9e9-11ea-8f54-0229b00d348b.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/93609489-8eff1780-f9e9-11ea-9f95-a7638dd6f4fd.png" width="300"/>

Update release notes:
